json-schema/json-schema/src/main/scala/endpoints/algebra/Codec.scala

```diff
@@ -68,4 +67,40 @@ object Codec {
       Decoder.sequentially(ab)(bc)
     )
 
+  /** Produce a codec that passes values through without touching them */
+  def identity[A]: Codec[A, A] = new Codec[A, A] {
+    def decode(from: A): Validated[A] = Valid(from)
+    def encode(from: A): A = from
+  }
+
+  /** Produce a codec to/from a string. An [[Invalid]] message mentioning the
+    * type name is produced when parsing fails.
+    *
+    * @param type name of the type being decoded
+    * @param parse parsing function to use
+    * @param print printing function to use
+    */
+  def tryParseString[A](
```

Comment:
I was considering making this method private. (See https://github.com/julienrf/endpoints/pull/537/files#diff-26b01f5a906ec9633108d28cfd1aabb1L161-R175)

What do you think?

Reply:
That is reasonable - it does seem a bit ad-hoc.

Somewhat ironically, my company _is_ using `tryParseString` in our codebase, so we may end up needing to duplicate some part of this function if it becomes private.

Let me know what you prefer and I'll adjust accordingly.

Reply:
We could leave it public but I would like to rename it to something that includes “catching”. For instance `parseStringCatching[A]`? And we should document the fact that exceptions thrown during parsing are caught and turn into an `Invalid` result.